traffic-dashboard/src/pages/mocks/add-mock-dialog/add-mock.dialog.tsx

```diff
@@ -0,0 +1,87 @@
+import {
+  Dialog,
+  Card,
+  TextField,
+  Typography,
+  Button,
+  CircularProgress,
+} from "@mui/material";
+import styles from "./add-mock-dialog.module.scss";
+import { useState } from "react";
+import { createMock } from "../../../api/api";
+
+type AddMockDialogProps = { open: boolean; close: () => void };
+
+export const AddMockDialog: React.FC<AddMockDialogProps> = ({
+  open,
+  close,
+}) => {
+  const [port, setPort] = useState<number>();
+  const [method, setMethod] = useState<string>("GET");
+  const [endpoint, setEndpoint] = useState<string>("");
+  const [status, setStatus] = useState<number>(200);
+  const [data, setData] = useState<unknown>("{ }");
```

Comment:
Maybe worth keeping it as an object and transforming to a string just before pasting it on a control. When referencing the data from the control we can process it into an object and thus also check whether the format is correct.

Reply:
Need to refactor it to use a form library